drivers/serial/uart_nrfx_uart.c

```diff
@@ -563,7 +565,10 @@ static void tx_isr(void)
 	} else {
 #if	defined(DT_NORDIC_NRF_UART_UART_0_RTS_PIN) && \
 		defined(DT_NORDIC_NRF_UART_UART_0_CTS_PIN)
-		k_delayed_work_cancel(&uart0_cb.tx_timeout_work);
+
+		if (uart0_cb.tx_timeout != K_FOREVER) {
```

Comment:
Right now there is some inconsistency in the way `k_delayed_work_cancel` is called. Sometimes it is with such check, sometimes without (see `uart_nrfx_tx_abort` and also calls related to `rx_timeout_work`). Maybe it would be good to clean it up? Perhaps these checks could be just omitted. At most `k_delayed_work_cancel` will return `-EINVAL`, what does not seem to be harmful.

Reply:
From kernel.h:
```
/**
 * @brief Cancel a delayed work item.
 [...]
 * @note The result of calling this on a k_delayed_work item that has
 * not been submitted (i.e. before the return of the
 * k_delayed_work_submit_to_queue() call) is undefined.
 [...]
 */
extern int k_delayed_work_cancel(struct k_delayed_work *work);
```
So we have to check if we don't cancel work that was not submitted. Added checks everywhere just for safety

Reply:
You're absolutely right. I was looking at the same documentation and I somehow overlooked this note...